arangod/Graph/Graph.cpp

```diff
@@ -189,6 +190,58 @@ void Graph::addVertexCollection(std::string const& name) {
   _vertexColls.emplace(name);
 }
 
+void Graph::rebuildOrphans(EdgeDefinition const& oldEdgeDefinition) {
+  // previous orphans may still be orphans...
+  std::set<std::string> orphans{orphanCollections()};
+
+  // previous vertex collections from the overwritten may be orphaned...
+  setUnion(orphans, oldEdgeDefinition.getFrom());
+  setUnion(orphans, oldEdgeDefinition.getTo());
+
+  // ...except they occur in any other edge definition, including the new one.
+  for (auto const& it : edgeDefinitions()) {
+    EdgeDefinition const& edgeDef = it.second;
+
+    setMinus(orphans, edgeDef.getFrom());
+    setMinus(orphans, edgeDef.getTo());
+  }
+  _orphanColls = orphans;
```

Comment:
Please iterate only over the new potential orphans and after validating that they are orphaned, insert the leftover ones into _orphanColls instead of overwriting.